commit 3edede08ff37c6a9370510508d5eeb54890baf47
Author: Justin Iurman <justin.iurman@uliege.be>
Date:   Tue Jul 20 21:42:59 2021 +0200

    ipv6: ioam: Support for IOAM injection with lwtunnels
    
    Add support for the IOAM inline insertion (only for the host-to-host use case)
    which is per-route configured with lightweight tunnels. The target is iproute2
    and the patch is ready. It will be posted as soon as this patchset is merged.
    Here is an overview:
    
    $ ip -6 ro ad fc00::1/128 encap ioam6 trace type 0x800000 ns 1 size 12 dev eth0
    
    This example configures an IOAM Pre-allocated Trace option attached to the
    fc00::1/128 prefix. The IOAM namespace (ns) is 1, the size of the pre-allocated
    trace data block is 12 octets (size) and only the first IOAM data (bit 0:
    hop_limit + node id) is included in the trace (type) represented as a bitfield.
    
    The reason why the in-transit (IPv6-in-IPv6 encapsulation) use case is not
    implemented is explained on the patchset cover.
    
    Signed-off-by: Justin Iurman <justin.iurman@uliege.be>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/include/uapi/linux/ioam6_iptunnel.h b/include/uapi/linux/ioam6_iptunnel.h
new file mode 100644
index 000000000000..bae14636a8c8
--- /dev/null
+++ b/include/uapi/linux/ioam6_iptunnel.h
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: GPL-2.0+ WITH Linux-syscall-note */
+/*
+ *  IPv6 IOAM Lightweight Tunnel API
+ *
+ *  Author:
+ *  Justin Iurman <justin.iurman@uliege.be>
+ */
+
+#ifndef _UAPI_LINUX_IOAM6_IPTUNNEL_H
+#define _UAPI_LINUX_IOAM6_IPTUNNEL_H
+
+enum {
+	IOAM6_IPTUNNEL_UNSPEC,
+	IOAM6_IPTUNNEL_TRACE,		/* struct ioam6_trace_hdr */
+	__IOAM6_IPTUNNEL_MAX,
+};
+
+#define IOAM6_IPTUNNEL_MAX (__IOAM6_IPTUNNEL_MAX - 1)
+
+#endif /* _UAPI_LINUX_IOAM6_IPTUNNEL_H */